### PR TITLE
feat(pos-app): add internal build variant ("WPay Dev")

### PR DESCRIPTION
**Pull request by ignaciosantise**
## What

Adds a second Expo build variant to **pos-app** (WPay Mobile POS) that installs alongside Production on the same device, is used for local development, and can be distributed to the team via TestFlight / Firebase. Mirrors the Expo-CNG variant setup already used by `wallets/rn_cli_wallet`; CI wiring follows `release-appkit.yaml`.

| Variant | id | display name | icon |
|---|---|---|---|
| production | `com.reown.mobilepos` | WPay | production |
| internal | `com.reown.mobilepos.internal` | **WPay Dev** | badged "DEV" |

To keep the shared reusable workflows untouched (they special-case the literal string `internal`), the variant is named `internal` end-to-end; the only user-facing "Dev" touch is the display name and badged icon.

## Changes
- **`app.config.js`** (new): dynamic config keyed on `APP_VARIANT` — switches `name`, `icon`, `scheme`, `ios.bundleIdentifier`.
- **`plugins/withAndroidVariants.js`** (new, replaces `scripts/withAndroidReleaseSigningConfig.js`): Gradle `internal` buildType + `internal`/`release` signing configs.
- **`plugins/withAndroidVariantIcons.js`** (new): overlays badged mipmaps + `strings.xml` (`app_name = "WPay Dev"`) into the `internal` buildType from `native-icons/`.
- **`assets/icons/{production,internal}/`**, **`native-icons/android/internal/`** (new).
- **`secrets.properties.mock`**, **`package.json`** (`android:internal`, `android:build:internal`, `ios:internal`, `ios:prod`), **`app.json`** plugin list.
- **`.github/workflows/release-pos.yaml`**: `release-type: [internal, production]` input; conditional bundle id / scheme (`WPayDev`) / app-variant / firebase / testflight / keystore.

## Screenshots
Screenshot 2026-08-07 at 11 53 02 AM
Screenshot 2026-08-07 at 11 53 43 AM

## Verification (local)
- **Android** `assembleInternal` → APK `com.reown.mobilepos.internal`, versionName `1.0.0-internal`, label **WPay Dev**, badged icon.
- **Android** `assembleRelease` regression → `com.reown.mobilepos` / WPay (release-signed). No regression.
- **iOS** `WPayDev` build → installs on simulator as `com.reown.mobilepos.internal` / **WPay Dev**.
- Lint clean · Jest 225/225 · workflow YAML valid.

## ⚠️ Follow-up required before the CI `internal` dispatch can ship
Infra must be provisioned (as GH Actions vars/secrets):
- Separate **App Store Connect app / Apple ID** for `com.reown.mobilepos.internal` + match provisioning profile (`reown-com/mobile-match`) → `POS_IOS_INTERNAL_APPLE_ID`.
- **Firebase app** for `com.reown.mobilepos.internal` → `POS_ANDROID_INTERNAL_FIREBASE_APP_ID`, `FIREBASE_POS_INTERNAL_URL`.
- `TESTFLIGHT_POS_INTERNAL_URL`, `WC_INTERNAL_KEYSTORE` / `WC_INTERNAL_KEYSTORE_NAME` (or reuse the prod upload keystore; set `WC_*_INTERNAL` = `WC_*_UPLOAD` in `ANDROID_SECRETS_FILE`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/release-pos.yaml [hunk 1]
```diff
@@ -16,6 +16,13 @@ on:
           - both
           - android
           - ios
+      release-type:
+        description: 'Release type'
+        required: true
+        type: choice
+        options:
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply .github/workflows/release-pos.yaml:24]: Intentionally keeping `internal` first — for this workflow internal *is* the desired default, so the UI pre-selecting it is wanted behavior. Made the default explicit in the sibling change (see the `release-type` input comment). Addressed in 259dd4ca.

File: .github/workflows/release-pos.yaml [hunk 2]
```diff
@@ -16,6 +16,13 @@ on:
           - both
           - android
           - ios
+      release-type:
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply .github/workflows/release-pos.yaml:19]: Fixed in 259dd4ca — dropped `required: true` and set `required: false` + `default: internal`, so automated dispatch callers that omit `release-type` no longer error (and now default to internal, our intended default rather than production). Also flipped every downstream ternary to test `== 'production'`, so any empty/omitted value (`variant-*` tag pushes, run-name label) resolves to internal; production now only happens on explicit selection.

File: dapps/pos-app/plugins/withAndroidVariants.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    /(release\s*\{[^}]*?)signingConfig\s+signingConfigs\.debug/s,
+    "$1signingConfig signingConfigs.release",
+  );
+
+  return buildGradle;
+}
+
+const withAndroidVariants = (config) =>
+  withAppBuildGradle(config, (cfg) => {
+    if (cfg.modResults.language === "groovy") {
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply dapps/pos-app/plugins/withAndroidVariants.js]: Fixed in 259dd4ca — replaced the silent no-op with a guard clause that `console.warn`s when `app/build.gradle` is not Groovy (Kotlin DSL), naming the skipped signing config and `internal` buildType so a future migration surfaces the gap instead of producing an unsigned/missing-variant APK.
